### PR TITLE
Set up workflow trigger for bcr-ui rebuild

### PR DESCRIPTION
**Pull request by hobofan**
Addresses https://github.com/bazel-contrib/bcr-ui/issues/20

For this to work a `BCR_UI_REPO_ACCESS_TOKEN` secret needs to be set by one of the repo maintainers.
I've included a comment describing what kind of access token needs to be generated (and the newly released fine-grained PATs seem great for it, but need to be enabled on the `bazel-contrib` org first).

### CODE REVIEW

**Comment by meteorcloudy** [comment]
Just to clarify, `BCR_UI_REPO_ACCESS_TOKEN` should be created by the maintainer of https://github.com/bazel-contrib/bcr-ui, is that correct?

**Comment by meteorcloudy** [comment]
Should I merge this after `https://github.com/bazel-contrib/bcr-ui` is ready?

**Review by meteorcloudy — APPROVED** [review]

**Comment by hobofan** [comment]
> Just to clarify, BCR_UI_REPO_ACCESS_TOKEN should be created by the maintainer of [bazel-contrib/bcr-ui](https://github.com/bazel-contrib/bcr-ui), is that correct?

The options for that are:
- A "classical" PAT which would need to be created by someone with write rights to the `bazel-contrib/bcr-ui` repository. Since those (even with only the required scopes here) provide quite a lot of access to APIs, they are usually only reluctantly handed out (has been point of discussion in past SIG meetings IIRC). I would be comfortable with providing one, if required.
- A fine-grained access token. With that someone can create an access token scoped to `bazel-contrib/bcr-ui` only. However to issue them, the mechanism of fine-grained access tokens (right now) needs to be enabled (and configured) by an admin of `bazel-contrib` (@alexeagle ?) first.

> Should I merge this after https://github.com/bazel-contrib/bcr-ui is ready?

The bcr-ui side of it is ready (the receiving end of the `repository_dispatch` is here: https://github.com/bazel-contrib/bcr-ui/blob/main/.github/workflows/trigger-via-bcr.yml).

**Comment by meteorcloudy** [comment]
Thanks for the explain! One more silly question, after we have the access token, how do we make it available to this github action?

**Comment by meteorcloudy** [comment]
I see, sounds great! @alexeagle Looks like only you have the permission to create any of the access token, let me know when it's available.

**Comment by alexeagle** [comment]
Okay I just did the enrollment for the new fine-grained personal access tokens, they can now be requested in the bazel-contrib org. I wonder if that would have made our lives much easier with the publish-to-bcr app...

Will try to create such a token as well, though as @hobofan wrote the thing it would be nice for GH actions to be triggered as him to give credit.

**Comment by alexeagle** [comment]
Screen Shot 2022-10-19 at 4 24 00 PM

I sent the token value to @meteorcloudy on Bazel Slack DM to add as a secret to this repo.

**Comment by meteorcloudy** [comment]
image
It should be available now, I'm merging this PR

**Comment by meteorcloudy** [comment]
image

Looks like it's working!

**Comment by meteorcloudy** [comment]
A year has passed, and I'm seeing failures like https://github.com/bazelbuild/bazel-central-registry/actions/runs/11436909160

@alexeagle Can you help re-generate the token?

**Pull request merged**
